Add smart contract to style-guide (#9190)
ethereum/ethereum-org-website: src/content/contributing/style-guide/index.md

The same rules we apply to proof-of-work are applicable to proof-of-stake, proof-of-authority, proof-of-humanity, proof-of-individuality, etc.

> ### Smart Contract {#smart-contract}

Reviewer: On ethereum.org we use title case for headings (except proper nouns :))

```suggestion
### Smart contract {#smart-contract}
```

> Smart contract should be capitalized at the beginning of a sentence. In any other instance, all letters should be lower case.

Reviewer: ```suggestion
Smart contract is a common noun and should only be capitalized at the beginning of a sentence. In any other instance, all letters should be lowercase.
```

Author: 'lower case' should be 'lowercase' on the 'Zero-knowledge' section also (as should be the 'is a common noun' part). Should I add those too, do I need to open another PR to do that, or can I do it on this one somehow?
